src/main.rs

```diff
@@ -33,10 +32,8 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     conn.flush()?;
 
     loop {
-        let event = conn.wait_for_event()?;
-        match event.response_type() {
-            CLIENT_MESSAGE_EVENT => {
-                let event: ClientMessageEvent = event.into();
+        match conn.wait_for_event()? {
+            Event::ClientMessage(event) => {
```

Comment:
Awesome! This was probably my only gripe with the x11rb API, I'm glad you changed it.